### ISSUE TITLE
Consume diagnostic messages by reports at once

### ISSUE THREAD

**Issue by bhirsz** [MarketSquare/robotframework-robocop#1315]
Currently we're consuming diagnostic messages one by one in reports:

```
    def report(self, diagnostic: Diagnostic) -> None:
        diagnostic.model = self.current_model  # TODO: Embed it into diagnostic where the report is raised
        for report in self.reports.values():
            report.add_message(diagnostic)
```

```
    def add_message(self, message: Diagnostic) -> None:
        self.issues.append(message)
```

Since we usually don't do anything extra with diagnostic messages at runtime and we only print summaries (or we don't use messages at all, ie version report) we can simply aggregate all diagnostic messages and consume it once. It will save a bit of time and memory.
